dkist/wcs/models.py

```diff
@@ -739,55 +739,109 @@ def varying_celestial_transform_from_tables(crpix: Union[Iterable[float], u.Quan
 
 
 class Ravel(Model):
-    array_shape = (0, 0)
-    n_inputs = 2
     n_outputs = 1
     _separable = False
+    _input_units_allow_dimensionless = True
 
-    def __init__(self, array_shape, order='C', **kwargs):
-        super().__init__(**kwargs)
+    @property
+    def n_inputs(self):
+        return len(self.array_shape)
```

Comment:
Hi @WilliamJamieson, yes. We currently have a 3D case. The extension to 3D is generalizeable to N dimensions, so that is what I implemented, which allows 2D and 3D to be handled in the same code and allows for future functionality beyond 3D, if needed.